src/PayID/pay-id-client.ts

```diff
@@ -86,13 +86,22 @@ export default class PayIdClient {
 
     const api = new DefaultApi(undefined, basePath)
 
-    const options = PayIdClient.makeOptionsWithAcceptTypes(
-      `application/${network}+json`,
-    )
+    const accept = `application/${network}+json`
+    const options = PayIdClient.makeOptionsWithAcceptTypes(accept)
 
     try {
-      const { data } = await api.resolvePayID(path, options)
-      // TODO(keefertaylor): make sure the header matches the request.
+      const result = await api.resolvePayID(path, options)
+
+      // Ensure the retrieved result matched the requested type.
+      const contentType = result.headers['content-type']
+      if (!contentType.includes(accept)) {
+        throw new PayIdError(
+          PayIdErrorType.UnexpectedResponse,
+          'Received bad Content-Type',
```

Comment:
I think this is fine. Doug would probably argue that if you ask me for BTC, and I give you XRP, then as long as I say that in the response header, everything is fine.

But at an SDK level, I think throwing an error is appropriate.

EDIT: Just saw the test you wrote below. Really stellar test case.